relay-kafka/src/producer/mod.rs

```diff
@@ -306,17 +330,20 @@ impl KafkaClient {
         message: &impl Message,
     ) -> Result<&str, ClientError> {
         let serialized = message.serialize()?;
-        #[cfg(feature = "schemas")]
-        self.schema_validator
-            .validate_message_schema(topic, &serialized)
-            .map_err(ClientError::SchemaValidationFailed)?;
+
+        if let SerializationOutput::Json(ref bytes) = serialized {
+            #[cfg(debug_assertions)]
```

Comment:
Does putting the `cfg` on the `if let` work?

```suggestion
        #[cfg(debug_assertions)]
        if let SerializationOutput::Json(ref bytes) = serialized {
```